lib/src/parser/recursive.dart

```diff
@@ -531,6 +539,111 @@ class RecursiveAstParser {
+      NgToken innerValue) {
+    var text = innerValue.lexeme;
+    var absoluteTextOffset = innerValue.offset;
+    var tokens = new RegExp(r'({{)|(}})');
+    var mustaches = <InterpolationAst>[];
+    int seenOpenMustache;
+
+    var scanner = new StringScanner(text);
+    while (!scanner.isDone) {
+      var position = scanner.position;
+      var match = tokens.firstMatch(scanner.rest);
+      if (match != null) {
+        var matchPosition = position + match.start;
+
+        if (match.group(1) != null) {
+          // '{{' found.
+          if (seenOpenMustache == null) {
+            seenOpenMustache = matchPosition;
+          } else {
+            // Second '{{' found before '}}' closes it.
+            var firstMustacheBegin = absoluteTextOffset + seenOpenMustache;
+            exceptionHandler.handle(new AngularParserException(
+              NgParserWarningCode.UNTERMINATED_MUSTACHE,
+              firstMustacheBegin,
+              '{{'.length,
+            ));
+
+            mustaches.add(new InterpolationAst.parsed(
+              _source,
+              new NgToken.interpolationStart(firstMustacheBegin),
+              new NgToken.interpolationValue(
+                  firstMustacheBegin + '{{'.length,
+                  text.substring(
+                      seenOpenMustache + '{{'.length, matchPosition)),
+              new NgToken.generateErrorSynthetic(
+                  absoluteTextOffset + matchPosition,
+                  NgTokenType.interpolationEnd),
+            ));
+            seenOpenMustache = matchPosition;
+          }
+        }
+
+        if (match.group(2) != null) {
+          if (seenOpenMustache != null) {
+            var mustacheBegin = absoluteTextOffset + seenOpenMustache;
+            var mustacheEnd = absoluteTextOffset + matchPosition;
+            mustaches.add(new InterpolationAst.parsed(
+              _source,
+              new NgToken.interpolationStart(mustacheBegin),
+              new NgToken.interpolationValue(
+                  mustacheBegin + '{{'.length,
+                  text.substring(
+                      seenOpenMustache + '{{'.length, matchPosition)),
+              new NgToken.interpolationEnd(mustacheEnd),
+            ));
+            seenOpenMustache = null;
+          } else {
+            // Found '}}' before any '{{'
+            var mustacheBegin = absoluteTextOffset + position;
+            var mustacheEnd = absoluteTextOffset + matchPosition;
+            exceptionHandler.handle(new AngularParserException(
+              NgParserWarningCode.UNOPENED_MUSTACHE,
+              mustacheEnd,
+              '}}'.length,
+            ));
+            mustaches.add(new InterpolationAst.parsed(
+              _source,
+              new NgToken.generateErrorSynthetic(
+                  mustacheBegin, NgTokenType.interpolationStart),
+              new NgToken.interpolationValue(
+                  mustacheBegin, text.substring(position, matchPosition)),
```

Comment:
After mulling this over, I think your suggested approach is the best way to resolve dangling mustaches. One of the largest pitfalls of mustaches in particular is that we're dealing with dart expressions and potentially recovering too hard can lead to many pitfalls. 

The alternative - not recovering but simply marking danglings - is a far safer option since the intended expressions will just be treated as plain text... which has no real consequences aside from the intended behavior not occurring (which is fine considering there is an error). 

However, the one exception to this will be if the enclosing space of a mustache is only whitespace, I'll treat that space as enclosing mustache. 

For example: 
`
<someTag someAttr="blah blah {{    ">
`
In this scenario, the space following the '{{' will be treated as '{{    ' since it's fairly safe to treat this space as mustache. It will provide significant benefit in terms of autocompletion since a user might have his/her text cursor in that space and may be in the process of adding content.

On the other hand: 
`
<someTag someAttr="blah blah {{ blah blah">
`
In this scenario, ' blah blah' follows the dangling mustache - which may or may not be intended to be part of the mustache. Assuming it is NOT is a safer option so the overall attribute value will be treated as plain text with the error thrown in indicating the dangling mustache.

This will be similar for dangling '}}'. 

HTML-text interpolation will have the one extra step of accommodating for newline:
`
<div>
    Age: {{
    Name: Bob
</div>
`
In this scenario, there will still be an interpolation at the end of the '{{' since it is a dangling mustache that encounters a newline. This means that a user can autocomplete if the cursor follows after the '{{' while on the same line. 

On the other hand:
`
<div>
    Age: {{ something here
    Name: Bob
</div>
`
In this scenario, the '{{ something here' will be treated as plain-text instead of a mustache expression until the user manually closes it.